VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -25,17 +25,924 @@ if ($PSEdition -eq 'Desktop') {
+        }
+        else {
+            Write-Host "ESXi Certificate Management Mode already set to $mode on the vCenter server $($vcfVcenterDetails.fqdn)"
+        }
+    }
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+    Finally {
+        Disconnect-VIServer $vcfVcenterDetails.fqdn -Confirm:$false -WarningAction SilentlyContinue
+    }
+}
+Export-ModuleMember -Function Set-CertManagementModeForESXi
+
+
+
+Function Get-vSANHealthSummary {
+
+    <#
+        .SYNOPSIS
+        Get the vSAN health summary from vCenter Server for given cluster 
+
+        .DESCRIPTION
+        This function gets the vSAN health summary from vCenter Server for a given cluster. If any status is YELLOW or RED, a WARNING or ERROR will be raised
+
+        .EXAMPLE
+        Get-vSANHealthSummary -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re123! -domain sfo-m01 -cluster sfo-m01-cl01 
+        This example gets the vSAN health summary for cluster sfo-m01-cl01 
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $domain,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $cluster
+
+    )
+    Try {
+        $vcfVcenterDetails = Get-vCenterServerConnection -server $server -user $user -pass $pass -domain $domain
+        $vSANClusterHealthSystem = Get-VSANView -Id "VsanVcClusterHealthSystem-vsan-cluster-health-system"
+        $cluster_view = (Get-Cluster -Name $cluster).ExtensionData.MoRef
+        $results = $vSANClusterHealthSystem.VsanQueryVcClusterHealthSummary($cluster_view, $null, $null, $true, $null, $null, 'defaultView')
+        $healthCheckGroups = $results.groups
+
+        foreach ($healthCheckGroup in $healthCheckGroups) {     
+            $health = @("Yellow", "Red")
+            $output = $healthCheckGroup.grouptests | Where-Object TestHealth -in $health | Select-Object TestHealth, @{l = "TestId"; e = { $_.testid.split(".") | Select-Object -last 1 } }, TestName, TestShortDescription, @{l = "Group"; e = { $healthCheckGroup.GroupName } }
+            $healthCheckTestHealth = $output.TestHealth
+            $healthCheckTestName = $output.TestName
+            $healthCheckTestShortDescription = $output.TestShortDescription
+            if ($healthCheckTestName) {
+                if ($healthCheckTestHealth -eq "yellow") {
+                    $overallStatus = ($overallStatus, 1 | Measure-Object -Max).Maximum
+                    Write-Warning "$vCenter - vSAN cluster $Cluster | vSAN Alarm Name - $healthCheckTestName | Alarm Description - $healthCheckTestShortDescription" 
```

Comment:
I don't see where `$vCenter` is coming from. Should it be `$($vcfVcenterDetails).fqdn`?

Use `$cluster`.